subsys/app_event_manager/Kconfig

```diff
@@ -12,6 +12,15 @@ menuconfig APP_EVENT_MANAGER
 
 if APP_EVENT_MANAGER
 
+config APP_EVENT_MANAGER_REBOOT_ON_EVENT_ALLOC_FAIL
+	bool "Reboot on event allocation failure"
+	depends on REBOOT
+	default y
```

Comment:
Given that `sys_reboot` does not flush logs, the `LOG_ERR` message will likely be lost before it is printed.

`k_panic` does flush logs, meaning the message will be visible before we reboot.

As such, I'd default this to `n`, as it's the least surprising option.
If an application wishes to "just" reboot, it can opt in.

Reply:
I prefer to keep it enabled by default to keep backwards compatibility (keep in mind that for debug configurations assertion failure would lead to log flush anyway), I can change it if needed, @pdunaj what's your opinion?

Reply:
I would go with backward compatibility in this case.

Reply:
@ball-hayden, is the approach acceptable for you?

Reply:
@ball-hayden, please approve the PR and we are ready to merge :)